crates/edr_provider/src/requests/serde.rs

```diff
@@ -377,35 +377,14 @@ impl<'de> Deserialize<'de> for Timestamp {
     where
         D: Deserializer<'de>,
     {
-        deserialize_timestamp_u64(deserializer).map(Timestamp)
-    }
-}
-
-fn deserialize_timestamp_u64<'de, D>(deserializer: D) -> Result<u64, D::Error>
-where
-    D: Deserializer<'de>,
-{
-    let error_msg = || {
-        serde::de::Error::custom(
-            "Timestamp must be a positive integer or a (`0x`-prefixed hex) string not exceeding 2^64.",
-        )
-    };
-
-    match serde_json::Value::deserialize(deserializer)? {
-        // Unlike other JSON-RPC types, timestamp can also be a number
-        serde_json::Value::Number(num) => num.as_u64().ok_or_else(error_msg),
-        serde_json::Value::String(ref value) => {
-            if let Some(value) = value.strip_prefix("0x") {
-                u64::from_str_radix(value, 16).map_err(|_err| error_msg())
-            } else {
-                // symmetrical and more explicit about the base 10
-                #[allow(clippy::from_str_radix_10)]
-                // Since it exceptionally can also be a number, be more relaxed
-                // and accept stringified numbers as well (base 10 by default).
-                u64::from_str_radix(value, 10).map_err(|_err| error_msg())
-            }
-        }
-        _ => Err(error_msg()),
+        // TODO: Accept only `0x`-prefixed hex strings (and possibly base 10 strings).
+        U64::deserialize(deserializer)
```

Comment:
@Wodann like we discussed, this uses `U64::deserialize` to keep using the old logic. We can revisit fixing this and other deserialization logic in a follow-up PR.